pkg/policy/visibility.go

```diff
@@ -127,13 +128,14 @@ func NewVisibilityPolicy(anno string) (*VisibilityPolicy, error) {
 	return nvp, nil
 }
 
-func generateL7AllowAllRules(parser L7ParserType) L7DataMap {
+func generateL7AllowAllRules(parser L7ParserType, namespace, pod string) L7DataMap {
 	var m L7DataMap
 	switch parser {
 	case ParserTypeDNS:
 		m = L7DataMap{}
 		// Create an entry to explicitly allow all at L7 for DNS.
 		emptyL3Selector := &identitySelector{source: &labelIdentitySelector{selector: api.WildcardEndpointSelector}, key: wildcardSelectorKey}
+		emptyL3Selector.metadataLbls = labels.LabelArray{{Key: "annotationSource", Value: strings.Join([]string{namespace, pod}, "/"), Source: labels.LabelSourceK8s}}
```

Comment:
Elsewhere we use this label here for visibility annotations, but it does not include the pod name and namespace, which I think was a missed opportunity:
https://github.com/cilium/cilium/blob/2f829958b714feae93fd65c8e39d6fdaf54c8a3f/pkg/policy/mapstate.go#L1096

Unless I'm confusing the semantics of the labels here, I think we should use a similar scheme to the labels we use for `CiliumNetwork` policies:

https://github.com/cilium/cilium/blob/ace97409a74d5ae6d57d8c0d4d9021bca559e039/pkg/k8s/apis/cilium.io/utils/utils.go#L51-L52

So it could be something like the following three labels:

```
        "k8s:io.cilium.k8s.policy.derived-from=PodVisibilityAnnotation",
        "k8s:io.cilium.k8s.policy.name=deathstar-api-protection",
        "k8s:io.cilium.k8s.policy.namespace=default",
```

Reply:
Do we have an issue for mapstate? I feel like this would be out of scope for this particular PR, but I think you are right in that it would be an improvement.

Will push out a change with revamped labels per your suggestion soon.

Reply:
I don't think we have an issue. I just noticed this while reviewing your PR. 

The original mapstate label was added a long time ago by myself https://github.com/cilium/cilium/pull/10512 - besides for debugging, I think it's only consumer is Hubble https://github.com/cilium/cilium/pull/27854